packages/editor-preview/src/browser/editor-preview-manager.ts

```diff
@@ -79,27 +81,34 @@ export class EditorPreviewManager extends WidgetOpenHandler<EditorPreviewWidget|
             } else {
                 this.shell.addWidget(editorWidget, {area: 'main'});
             }
-            preview.close();
+            preview.dispose();
             this.shell.activateWidget(editorWidget.id);
-            this.currentEditorPreview = undefined;
+            this.currentEditorPreview = Promise.resolve(undefined);
         });
     }
 
-    protected isCurrentPreviewUri(uri: URI): boolean {
-        const currentUri = this.currentEditorPreview && this.currentEditorPreview.getResourceUri();
+    protected async isCurrentPreviewUri(uri: URI): Promise<boolean> {
+        const editorPreview = await this.currentEditorPreview;
+        const currentUri = editorPreview && editorPreview.getResourceUri();
         return !!currentUri && currentUri.isEqualOrParent(uri);
     }
 
     canHandle(uri: URI, options?: PreviewEditorOpenerOptions): MaybePromise<number> {
-        if (this.preferences['editor.enablePreview'] && (options && options.preview || this.isCurrentPreviewUri(uri))) {
-            return 200;
-        }
-        return 0;
+        return this.isCurrentPreviewUri(uri).then(isCurrentPreviewUri => {
+            if (this.preferences['editor.enablePreview'] && (options && options.preview || isCurrentPreviewUri)) {
+                return 200;
+            }
+            return 0;
+        });
     }
 
     async open(uri: URI, options?: PreviewEditorOpenerOptions): Promise<EditorPreviewWidget | EditorWidget> {
         options = {...options, mode: 'open'};
 
+        const deferred = new Deferred<EditorPreviewWidget | undefined>();
+        const previousPreview = await this.currentEditorPreview;
```

Comment:
can `currentEditorPreview` be undefined here?

Reply:
Yes, it can be undefined, but it should be OK.